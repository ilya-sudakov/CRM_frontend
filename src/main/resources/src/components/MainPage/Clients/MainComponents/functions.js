import { exportClientsEmailsCSV } from '../../../../utils/xlsxFunctions.jsx'
import { getClients } from '../../../../utils/RequestsAPI/Clients.jsx'

export const sortClients = (clients, searchQuery, sortOrder) => {
  return clients.sort((a, b) => {
    if (searchQuery !== '') {
      if (sortOrder.curSort === 'nextDateContact') {
        if (new Date(a[sortOrder.curSort]) < new Date(b[sortOrder.curSort])) {
          return sortOrder[sortOrder.curSort] === 'desc' ? 1 : -1
        }
        if (new Date(a[sortOrder.curSort]) > new Date(b[sortOrder.curSort])) {
          return sortOrder[sortOrder.curSort] === 'desc' ? -1 : 1
        }
        return 0
      } else {
        if (a[sortOrder.curSort] < b[sortOrder.curSort]) {
          return sortOrder[sortOrder.curSort] === 'desc' ? 1 : -1
        }
        if (a[sortOrder.curSort] > b[sortOrder.curSort]) {
          return sortOrder[sortOrder.curSort] === 'desc' ? -1 : 1
        }
        return 0
      }
    }
  })
}

export const getEmailsExcel = () => {
  let totalClients = 1
  let clients = []
  return getClients(1)
    .then((res) => res.json())
    .then((res) => {
      totalClients = res.totalElements
      return getClients(totalClients)
    })
    .then((res) => res.json())
    .then((res) => {
      clients = res.content
      console.log(clients)
      exportClientsEmailsCSV(clients)
    })
    .catch((error) => {
      console.log(error)
    })
}
