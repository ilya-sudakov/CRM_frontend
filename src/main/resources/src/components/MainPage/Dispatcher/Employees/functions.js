export const filterEmployeesBySearchQuery = (data, searchQuery) => {
  const query = searchQuery.toLowerCase();
  return data.filter((item) => {
    if (item.name === null) return false;
    const isFound =
      item.lastName.toLowerCase().includes(query) ||
      item.name.toLowerCase().includes(query) ||
      item.middleName.toLowerCase().includes(query) ||
      item.id.toString().includes(query) ||
      (item.yearOfBirth && item.yearOfBirth.toString().includes(query)) ||
      (item.dateOfBirth && item.dateOfBirth.toString().includes(query)) ||
      item.citizenship.toLowerCase().includes(query) ||
      item.workshop.toLowerCase().includes(query) ||
      item.position.toLowerCase().includes(query) ||
      item.comment.toLowerCase().includes(query) ||
      item.relevance.toLowerCase().includes(query);
    return isFound;
  });
};
