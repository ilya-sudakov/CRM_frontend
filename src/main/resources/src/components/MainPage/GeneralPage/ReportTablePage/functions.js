export const getDaysArray = (start, end) => {
  let arr = [];
  for (let dt = new Date(start); dt <= end; dt.setDate(dt.getDate() + 1)) {
    arr.push(new Date(dt));
  }
  return arr;
};

export const sortEmployees = (employees) => {
  return employees.sort((a, b) => {
    if (a.employee.lastName < b.employee.lastName) {
      return -1;
    }
    if (a.employee.lastName > b.employee.lastName) {
      return 1;
    }
    return 0;
  });
};

export const getMonthDates = (curDate) => {
  const date =
    new Date(curDate.getFullYear(), curDate.getMonth() + 1, 0).getDate() + 1;
  //Получаем массив с датами месяца
  let dates = [];
  for (let i = 1; i < date; i++) dates.push(i);

  return dates;
};
