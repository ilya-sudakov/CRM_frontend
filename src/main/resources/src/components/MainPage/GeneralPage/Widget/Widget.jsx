import React from 'react'
import PropTypes from 'prop-types'
import './Widget.scss'
import Button from '../../../../utils/Form/Button/Button.jsx'
import { useHistory } from 'react-router-dom'

const Widget = ({
  title = '',
  className = '',
  content = null,
  linkTo = {
    text: '',
    address: '/',
  },
  subTitle = '',
  customHeader = null,
}) => {
  let history = useHistory()

  return (
    <div className={`widget ${className ?? ''}`}>
      {customHeader ?? (
        <div className="widget__title">
          <div className="widget__sub-title">{subTitle}</div>
          <span>{title}</span>
          {linkTo ? (
            <Button
              className="main-window__button main-window__button--inverted"
              inverted
              text={linkTo.text}
              imgSrc={linkTo.img ?? null}
              onClick={() => history.push(linkTo.address)}
            ></Button>
          ) : null}
        </div>
      )}
      <div className="widget__content">{content}</div>
    </div>
  )
}

export default Widget

Widget.proptypes = {
  title: PropTypes.string,
  className: PropTypes.string,
  linkTo: PropTypes.object,
  subTitle: PropTypes.string,
  customHeader: PropTypes.object,
}
