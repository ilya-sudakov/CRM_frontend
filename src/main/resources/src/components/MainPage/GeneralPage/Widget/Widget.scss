@import "../../../../variables.scss";

.widget {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
    box-sizing: border-box;
    position: relative;
    width: calc(100% - 40px);
    box-shadow: 0 1px 3px rgba(59, 53, 53, 0.12), 0 1px 5px rgba(0, 0, 0, 0.1);
    background-color: $secondary-text-color;
    border-radius: 5px;
    padding: 9px 26px;
    padding-top: 15px;
    min-width: 300px;
    min-height: 298px;
    max-height: 316px;
    margin-bottom: 15px;
    margin-right: 15px;
    max-width: 600px;

    &__title {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        justify-content: flex-start;
        flex-wrap: wrap;
        align-items: center;
        font-size: 1.25rem;
        position: relative;
        width: 100%;
        margin-bottom: 5px;

        .main-window__button {
            padding: 5px 10px;
            margin-top: -1.5rem;
            font-size: 70%;
            margin-left: auto;
        }

        .widget__sub-title {
            font-size: 65%;
            color: #666666;
            width: 100%;
        }
    }

    &__content {
        width: 100%;
        overflow-y: auto;
    }

    @media (max-width: 768px) {
        max-width: none;
        width: 100%;
        margin-right: 0;
    }
}
