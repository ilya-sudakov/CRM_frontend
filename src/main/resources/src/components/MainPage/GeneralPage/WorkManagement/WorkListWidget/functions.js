export const sortEmployeesObject = (employees) => {
  return employees.sort((a, b) => {
    if (a[1].lastName < b[1].lastName) {
      return -1;
    }
    if (a[1].lastName > b[1].lastName) {
      return 1;
    }
    return 0;
  });
};

export const filterEmployeesObject = (employees, workshop) => {
  return employees.filter((employee) => {
    const item = employee[1];
    if (item.workshop === workshop.name) return true;
  });
};

export const getAllEmployees = (works) => {
  let newEmployees = {};
  works.map((work) => {
    if (newEmployees[work.employee.id] === undefined) {
      return (newEmployees = Object.assign({
        ...newEmployees,
        [work.employee.id]: work.employee,
      }));
    }
  });
  return newEmployees;
};
