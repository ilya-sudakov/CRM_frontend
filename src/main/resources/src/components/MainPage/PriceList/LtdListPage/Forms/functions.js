import React from "react";
import InputText from "../../../../../utils/Form/InputText/InputText.jsx";
import {
  getBIKByINN,
  getInfoByINN,
} from "../../../../../utils/RequestsAPI/Clients.jsx";

export const fetchINNData = (
  inputs,
  setIsLoading,
  setFormInputs,
  validateField
) => {
  setIsLoading(true);
  let formData = {};
  //Получаем данные о компании(Головной офис - MAIN BRANCH) по ИНН
  getInfoByINN({ query: inputs.inn, branch_type: "MAIN" })
    .then((res) => res.json())
    .then((res) => {
      if (res.suggestions.length > 0) {
        console.log(res);
        const data = res.suggestions[0].data;
        formData = Object.assign({
          ...inputs,
          name: data.name.full,
          shortName: data.name.short,
          kpp: data.kpp,
          okpo: data.okpo,
          okved: data.okved,
          ogrn: data.ogrn,
          legalAddress: data.address.value,
          generalDirector: data.management?.name,
        });
        validateField("name", data.name.full);
        validateField("shortName", data.name.short);
        validateField("kpp", data.kpp);
        validateField("okpo", data.okpo);
        validateField("okved", data.okved);
        validateField("ogrn", data.ogrn);
        validateField("legalAddress", data.legalAddress);
        validateField("generalDirector", data.generalDirector);
        return formData;
      } else return null;
    })
    .then((newData) => {
      if (newData !== null) {
        //Получаем БИК банка по названию компании
        getBIKByINN({ query: formData.name })
          .then((res) => res.json())
          .then((res) => {
            console.log(res);
            setIsLoading(false);
            validateField(
              "bik",
              res.suggestions.length > 0 ? res.suggestions[0].data.bic : ""
            );
            setFormInputs({
              ...formData,
              bik:
                res.suggestions.length > 0 ? res.suggestions[0].data.bic : "",
            });
            setIsLoading(false);
          });
      } else {
        alert("Не найдено данных с данным ИНН");
        setIsLoading(false);
      }
    });
};

export const ltdFormCreateInput = (options, errors, inputs) => {
  return (
    <InputText
      inputName={options.inputName}
      required={options.required ?? false}
      error={options.required ? errors.formErrors[options.name] : undefined}
      defaultValue={inputs.formInputs[options.name]}
      handleInputChange={({ target }) =>
        inputs.handleInputChange(options.name, target.value)
      }
      errorsArr={options.required ? errors.formErrors : undefined}
      setErrorsArr={options.required ? errors.setFormErrors : undefined}
    />
  );
};
