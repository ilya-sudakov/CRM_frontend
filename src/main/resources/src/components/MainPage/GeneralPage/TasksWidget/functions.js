import { getMainTasks } from '../../../../utils/RequestsAPI/MainTasks.jsx'
import { formatDateStringNoYear } from '../../../../utils/functions.jsx'

export const filterTasks = (tasks) => {
  return tasks.filter((task) => task.condition !== 'Выполнено')
}

export const filterTasksByUser = (tasks, userRole = 'admin') => {
  return tasks.filter((task) => task.responsible === userRole)
}

export const getTasksList = async () => {
  let tasks = []
  await getMainTasks()
    .then((res) => res.json())
    .then((res) => {
      return (tasks = res)
    })
    .catch((err) => {
      return console.error(err)
    })
  return tasks
}

export const getTasksControlDatesList = (tasks) => {
  let controlDates = {}
  tasks
    .sort((a, b) => {
      if (a.dateControl < b.dateControl) {
        return -1
      }
      if (a.dateControl > b.dateControl) {
        return 1
      }
      return 0
    })
    .map((task) => {
      const curDate = formatDateStringNoYear(task.dateControl)

      return (controlDates = {
        ...controlDates,
        [curDate]:
          controlDates[curDate] === undefined
            ? [task]
            : [...controlDates[curDate], task],
      })
    })
  return controlDates
}
