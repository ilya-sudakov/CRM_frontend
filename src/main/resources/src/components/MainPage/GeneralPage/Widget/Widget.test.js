import React from "react";
import Widget from "./Widget.jsx";
import { render, cleanup, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { renderWithRouter } from "../../../../utils/testing/functions.js";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe("Widget component", () => {
  afterEach(cleanup);

  it("matches snapshot", () => {
    const { asFragment } = render(<Widget />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders w/o props", () => {
    renderWithRouter(<Widget />);
  });

  it("redirects on link button click", () => {
    renderWithRouter(
      <Widget
        linkTo={{
          text: "Перейти",
          address: "/requests/open",
        }}
      />
    );
    fireEvent.click(screen.getByRole("button"));
    expect(mockHistoryPush).toHaveBeenCalledWith("/requests");
  });

  it("renders content", async () => {
    renderWithRouter(<Widget content="Pirog" />);
    expect(await screen.findByText(/Pirog/)).toBeInTheDocument();
  });
});
