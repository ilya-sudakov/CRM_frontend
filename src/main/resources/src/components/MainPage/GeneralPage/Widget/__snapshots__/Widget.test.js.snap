// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="widget "
  >
    <div
      class="widget__title"
    >
      <div
        class="widget__sub-title"
      />
      <span />
    </div>
    <div
      class="widget__content"
    />
  </div>
</DocumentFragment>
`;
